database/memorydb.go

```diff
@@ -0,0 +1,81 @@
+package database
+
+import (
+	"fmt"
+	"sync"
+)
+
+type todos map[int]string
+
+type Database struct {
+	mu     sync.Mutex
+	data   todos
+	lastId int
+}
+
+func CreateDatabase() *Database {
+	return &Database{data: make(todos), lastId: -1}
+}
+
+func (db *Database) FetchTodo(id int) (string, error) {
+	var err error
+	db.mu.Lock()
+	defer db.mu.Unlock()
+
+	val, exists := db.data[id]
+
+	if !exists {
+		err = fmt.Errorf("ID: %v doesn't exist", id)
+	}
+
+	return val, err
+}
+
+func (db *Database) FetchAll() map[int]string {
+	db.mu.Lock()
+	defer db.mu.Unlock()
+	todoList := make(map[int]string, len(db.data))
+
+	for k, v := range db.data {
+		todoList[k] = v
+	}
+
+	return todoList
+}
+
+func (db *Database) CreateTodo(todoText string) int {
```

Comment:
There's no need to return anything here. If you have that need you need to decouple your code

Reply:
fixed 75db70ae0fa1a97da23f19d14057ae3cb27572d4

Reply:
The reason I thought that would be ok to return it was to be able to return the ID of the new TODO, so it could be updated, fetched or removed. 
Which could be a good approach to do it? 
or it's not needed at all since fetch all will be returning `ID: TODO`

Reply:
Now that you have refactored your code, it has a place for it, well done